lowranceusr.cc

```diff
@@ -489,167 +1230,320 @@ lowranceusr_parse_icons()
-    trk_head = route_head_alloc();
-    trk_head->rte_num = ++trk_num;
-    track_add_head(trk_head);
+  if (global_opts.debug_level > 1) {
+    printf(MYNAME " parse_trails: Trail '%s'\n", qPrintable(trk_head->rte_name));
+  }
 
-    /* trail name */
-    int text_len = lowranceusr_readstr(&buff[0], MAXUSRSTRINGSIZE, file_in);
+  /* visible */
+  (void) gbfread(&buff[0], 1, 1, file_in);
+  /* num trail points */
+  short num_trail_points = gbfgetint16(file_in);
 
-    if (global_opts.debug_level >= 1) {
-      printf(MYNAME " parse_trails: name text len = %d\n", text_len);
-    }
+  if (global_opts.debug_level > 1) {
+    printf(MYNAME " parse_trails: Num Trail Points = %d\n", num_trail_points);
+  }
 
-    if (text_len) {
-      buff[text_len] = '\0';
-      trk_head->rte_name = buff;
-    }
+  /* max trail size */
+  int itmp = gbfgetint16(file_in);
 
-    if (global_opts.debug_level >= 1) {
-      printf(MYNAME " parse_trails: trail name = %s\n", qPrintable(trk_head->rte_name));
-    }
+  if (global_opts.debug_level > 1) {
+    printf(MYNAME " parse_trails: Max Trail size = %d\n", itmp);
+  }
 
-    /* visible */
-    (void) gbfread(&buff[0], 1, 1, file_in);
-    /* num trail points */
-    short num_trail_points = gbfgetint16(file_in);
+  if (num_trail_points) {
 
-    if (global_opts.debug_level >= 1) {
-      printf(MYNAME " parse_trails: num trail points = %d\n", num_trail_points);
-    }
+    while (num_trail_points && !gbfeof(file_in)) {
+      /* num section points */
+      short int num_section_points = gbfgetint16(file_in);
 
-    /* max trail size */
-    int itmp = gbfgetint16(file_in);
+      if (global_opts.debug_level > 1) {
+        printf(MYNAME " parse_trails: Num Section Points = %d\n", num_section_points);
+      }
 
-    if (global_opts.debug_level >= 1) {
-      printf(MYNAME " parse_trails: max trail size = %d\n", itmp);
+      for (int j = 0; j < num_section_points && !gbfeof(file_in); j++, num_trail_points--) {
+        Waypoint* wpt_tmp = new Waypoint;
+        wpt_tmp->latitude = lat_mm_to_deg(gbfgetint32(file_in));
+        wpt_tmp->longitude = lon_mm_to_deg(gbfgetint32(file_in));
+        /* continuous */
+        (void) gbfread(&buff[0], 1, 1, file_in);
+        if (!buff[0] && opt_seg_break && j) {
+          /* option to break trails into segments was specified */
+          auto trk_tmp = route_head_alloc();
+          trk_tmp->rte_num = ++(*trail_num);
+          trk_tmp->rte_name = trk_head->rte_name;
+          track_add_head(trk_tmp);
+          trk_head = trk_tmp;
+        }
+        track_add_wpt(trk_head, wpt_tmp);
+
+        if (global_opts.debug_level > 2) {
+          printf(MYNAME " parse_trails: Trail pt lat %f lon %f\n", wpt_tmp->latitude, wpt_tmp->longitude);
+        }
+      }
     }
+  } else {
+    /* remove the trail since it's empty */
+    track_del_head(trk_head);
+  }
+}
+
+static void
+lowranceusr4_parse_trail(int *trail_num)
```

Comment:
Left this in for consistency with lowranceusr_parse_trail().  Like lowranceusr_parse_trail() this may need to update if I find that trails need to be broken into segments.